juju/juju
# Remove all logs and infrastructure from mongo

Logs are now streamed from files on disk so all logging is removed from mongo. A previous PR implement this; this PR just removes the now unused db log code.
A few collection stats functions from deleted files in state were copied elsewhere as they are still used.

The agent only params to tune the log buffer are retained and renamed. We can decide on default values moving forward.

The migration log transfer LastLogTime was modified to scan the log file and parse the timestamp from there. The transfer then skips over log lines before that timestamp. This all replaces the state based log tracker. When the log lines are converted to json in the next PR, we'll get proper precision on the timestamps.

Finally, the syslog worker and syslog logging output destination is removed as we don't want to support it any more.

## QA steps

bootstrap and double check debug log works
Do a migration. before starting the migration,  I copied across to the target controller come log lines from the model being migrated. After migration, check the model log file to see that the original records are retained and new ones copied across.

## Links

**Jira card:** JUJU-5505
